src/model/mod.rs

```diff
@@ -1 +1,290 @@
+        .read_line(&mut buffer)
+        .expect("something went horribly wrong...");
+
+    /* Discard trailing newline */
+    let new_len = buffer.trim_right().len();
+    buffer.truncate(new_len);
+
+    buffer
+}
+
+/// The actual placement of the ships.
+fn place(player: &mut types::Player, ship: &types::ShipType) -> Result<(), String> {
+
+    /* Vector to collect the indices for the possible ship position. */
+    let mut indices = Vec::new();
+
+    /* 100 --> invalid value */
+    let mut input = 100;
+    while input == 100 {
+        input = get_input();
+        match player.own_board[input] {
+            types::SubField::Water => { indices.push(input); },
+            _ => {
+                println!("Invalid input, again please.");
+                input = 100;
+            },
+        }
+    }
+
+    println!("{}{}", "Enter 'h' for a horizontal (upwards) orientation of the ship,",
+        "'v' for a vertical (rightwards) one.");
+    /* x --> invalid value */
+    let mut orientation = "x".to_string();
+    while orientation == "x" {
+        orientation = read_string();
+        if orientation != "h" && orientation != "v" {
+            println!("Invalid input, again please.");
+            orientation = "x".to_string();
+        }
+    }
+
+    if orientation == "v" {
+        for _ in 0..ship.size - 1 {
+            /* Check to prevent out of bounds errors. */
+            if input >= 10 {
+                input -= 10;
+            }
+            /* input < 10 --> no field above */
+            if player.own_board[input] != types::SubField::Water || input < 10 {
+                return Err("Invalid position for this ship, please choose another coordinate."
+                    .to_string())
+            }
+            indices.push(input);
+        }
+    } else {
+        for _ in 0..ship.size - 1 {
+            /* Check to prevent out of bounds errors. */
+            if input < 99 {
+                input += 1;
+            }
+            /* unit position == 9 --> no field to the right */
+            if player.own_board[input] != types::SubField::Water || (input % 10) == 9 {
+                return Err("Invalid position for this ship, please choose another coordinate."
+                    .to_string())
+            }
+            indices.push(input);
+        }
+    }
+
+    /* Places the ships on the board. */
+    for i in indices.iter() {
+        player.own_board[*i] = types::SubField::Ship;
+    }
+
+    Ok(())
+}
+
+/// Handles the initial ship placement for each player.
+fn place_ships(mut player1: &mut types::Player, mut player2: &mut types::Player) {
+
+    /* A vector of all the ships each player needs to place.
+       The default version: #  Class of ship Size
+                            1   Carrier       5
+                            2   Battleship    4
+                            3   Cruiser       3
+                            4   Submarine     3
+                            5   Destroyer     2
+    */
```

Comment:
I took it from [Battleship_(game)](https://en.wikipedia.org/wiki/Battleship_(game))  but you're right, I also thought that there's way too much to hit on the board.. Changed it to the configuration from `default.ships`.